apps/meteor/app/reactions/server/setReaction.js

```diff
@@ -71,7 +72,7 @@ async function setReaction(room, user, message, reaction, shouldReact) {
 			}
 		}
 		callbacks.run('unsetReaction', message._id, reaction);
-		callbacks.run('afterUnsetReaction', message, { user, reaction, shouldReact });
+		callbacks.run('afterUnsetReaction', message, { user, reaction, shouldReact, oldMessage });
```

Comment:
Why is a second object with the same data needed here?

Reply:
It's not the exact same object, it looks like it is, but it is not. The `message` object is being changed here: https://github.com/RocketChat/Rocket.Chat/pull/26705/files#diff-559e52371075a947a366e0b700a3253057c65b680c6ba48c8cda135314d5b8d4R61, and to be able to handle correctly on the Matrix side, we do need the whole object. 😬